ampersand-chart.js

```diff
@@ -613,12 +632,27 @@
       if (this.model.drawXAxisLabels) {
         container.append('text')
           .attr('class', 'ampersand-graph-label')
+          .style('display', 'none')
+          .style('opacity', 0)
           .attr('y', height - 46)
           .attr('dy', '1.25em');
 
-        containers.select('text.ampersand-graph-label')
-          .transition()
-          .attr('x', ((barWidth * values.length) + barMargin * (values.length - 1)) / 2);
+        _.defer(function(containers, barWidth, values, barMargin, label) {
+          containers.select('text.ampersand-graph-label')
+            .style('opacity', 0)
+            .style('display', function(d, i) {
+              var width = Math.ceil(_.max(data, function(item) { return item[label].toString().length; })[label].toString().length * 6.25);
```

Comment:
This is a good example where `_.reduce` works well:
```js
var width = Math.ceil(_.reduce(data, function(max, item) { return Math.max(max, item[label].toString().length); }) * 6.25);
```

This avoids having to access `item[label].toString().length` multiple times and gives you a bit more control over what happens in the empty list scenario.